include/fms_platform.h

```diff
@@ -80,39 +52,28 @@ use,intrinsic :: iso_c_binding, only: c_double,c_float,c_int64_t, &
 !DEC$ MESSAGE:'Using cray pointers.'
 #endif
 
-
-!Control size of integers that will hold address values.
-!Appears for legacy reasons, but seems rather dangerous.
-#ifdef _32bits
-#define POINTER_KIND 4
-!DEC$ MESSAGE:'Using 4-byte addressing'
-#endif
-
-
-!If you do not want to use 64-bit integers.
-#ifdef no_8byte_integers
-#define LONG_KIND INT_KIND
-#endif
-
-
-!If you do not want to use 32-bit floats.
-#ifdef no_4byte_reals
-#define FLOAT_KIND DOUBLE_KIND
-#define NF_GET_VAR_REAL nf_get_var_double
-#define NF_GET_VARA_REAL nf_get_vara_double
-#define NF_GET_ATT_REAL nf_get_att_double
-#undef OVERLOAD_R4
-#undef OVERLOAD_C4
-#endif
-
-
-!If you want to use quad-precision.
-#define NO_QUAD_PRECISION
-#if defined(QUAD_PRECISION) && !defined(__PGI)
-#undef NO_QUAD_PRECISION
+!Set type kinds.
+#ifdef PORTABLE_KINDS
+use,intrinsic :: iso_fortran_env, only: real128
+use,intrinsic :: iso_c_binding, only: c_double,c_float,c_int64_t, &
+                                      c_int32_t,c_int16_t,c_intptr_t
+#define QUAD_KIND real128
```

Comment:
How are you dealing with the real128 value when the platform doesn't have support and sets QUAD_KIND to a negative number?

"Kind type parameters to specify a REAL type with a storage size of 32, 64, and 128 bits. It is negative if a target platform does not support the particular kind. (Fortran 2008 or later.)" - http://fortranwiki.org/fortran/show/iso_fortran_env

Reply:
There will be a test for HAVE_QUAD_PRECISION soon.  It isn't in place yet.

Currently, there is no use (as far as I can tell) of QUAD_KIND in the model.  The only places I have found of `NO_QUAD_PRECISION` use `selected_real_kind` to determine the precision.  

For now, the user will need to define `-DNO_QUAD_PRECISION` when compiling until autotools can set that automatically for the user.

Reply:
If it is not used, can we just not support quad precision?

Reply:
QP is actually used in various places that benefit from extended precision such as:  interpolations, grid metric calculations, etc.  As Seth said, we are not using the QUAD_KIND identifier, but instead relying on the use of a macro to gate platform support.

Reply:
To turn this on in configure.ac:
 `  AC_DEFINE([NO_QUAD_PRECISION], [1], [if true, no quad precision])`